includes/class-wc-cart.php

```diff
@@ -1938,4 +1938,15 @@ private function reset_totals() {
 		$this->fees_api->remove_all_fees();
 		do_action( 'woocommerce_cart_reset', $this, false );
 	}
+
+	/**
+	 * Returns the hash based on cart contents.
+	 *
+	 * @return string hash for cart content
+	 */
+	public function get_cart_hash() {
+		$hash = $this->session->get_cart_for_session() ? md5( json_encode( $this->session->get_cart_for_session() ) ) : '';
+
+		return apply_filters( 'woocommerce_add_to_cart_hash', $hash, $this->get_cart_for_session() );
```

Comment:
`$this->get_cart_for_session()` is deprecated and moved to `$this->session->get_cart_for_session()`.

Reply:
Done! Another question at this point: How to deprecate a filter name backward compatible? I would like to rename the filter to `woocommerce_cart_hash` for a more consistent filter naming.